WordPress/Classes/Services/StoreFacade.swift

```diff
@@ -0,0 +1,160 @@
+import Foundation
+import StoreKit
+
+enum ProductRequestError: ErrorType {
+    /// One of the requested product identifiers wasn't included in the response.
+    case MissingProduct
+
+    /// A product price couldn't be formatted into a String using the returned locale.
+    case InvalidProductPrice
+}
+
+protocol StoreFacade {
+    func getProductsWithIdentifiers(identifiers: Set<String>, success: Products -> Void, failure: ErrorType -> Void)
+}
+
+extension StoreFacade {
+    /// Requests prices for the given plans.
+    ///
+    /// On success, it calls the `success` function with an array of prices. If
+    /// one of the plans didn't have a product identifier, it's treated as a
+    /// "free" plan and the returned price will be an empty string.
+    func getPricesForPlans(plans: [Plan], success: [String] -> Void, failure: ErrorType -> Void) {
+        let identifiers = Set(plans.flatMap({ $0.productIdentifier }))
+        getProductsWithIdentifiers(
+            identifiers,
+            success: { products in
+                do {
+                    let prices = try plans.map({ plan -> String in
+                        return try priceForPlan(plan, products: products)
+                    })
+                    success(prices)
+                } catch let error {
+                    failure(error)
+                }
+            },
+            failure: failure
+        )
+    }
+}
+
+class StoreKitFacade: StoreFacade {
+    func getProductsWithIdentifiers(identifiers: Set<String>, success: Products -> Void, failure: ErrorType -> Void) {
+        let request = SKProductsRequest(productIdentifiers: identifiers)
```

Comment:
As we just discussed, the docs for `SKProductsRequest` states: 

> Be sure to keep a strong reference to the request object; otherwise, the system might deallocate the request before it can complete.

So I think the request will need retaining somewhere.